File: JavaScript/leetCodeDaily/3174_prblm.js
var clearDigits = function (s) {
  let stack = []; // Using an array as a stack

  for (let char of s) {
    if (/\d/.test(char)) {
      // If it's a digit, remove the last added character
      stack.pop();
    } else {
      stack.push(char); // Push valid characters
    }
  }
  return stack.join(""); // Convert array back to string
};

// Test case
console.log(clearDigits("abc3de5fg")); // Output: "abdfg"

File: C++/leet_Code_Daily/3174_prblm.cpp
#include <iostream>
using namespace std;

class Solution
{
public:
  string clearDigits(string s)
  {
    int j = 0; // Acts as the position for valid characters
    for (char c : s)
    {
      if (isdigit(c))
      {
        // If it's a digit, remove the last added character
        j = max(0, j - 1);
      }
      else
      {
        s[j++] = c; // Place valid characters at the correct position
      }
    }
    return s.substr(0, j); // Return the valid part of the string
  }
};

// Driver code
int main()
{
  Solution sol;
  string s = "abc3de5fg";             // Example Input
  cout << sol.clearDigits(s) << endl; // Output: "abdfg"
  return 0;
}
